### PR TITLE
Upgrade advisory schema to v2

### PR DESCRIPTION
**Pull request by luhring**
This PR introduces formality to the concept of a "schema" for advisory documents, and declares a new schema, version `2`, that's now used throughout the advisory-related functions of `wolfictl`.

⚠️  **THIS IS A BREAKING CHANGE.** As of this PR, `wolfictl` is no longer able to operate on data using the "v1" (original) data format used in advisory documents.

Commands updated to v2 schema:

- `wolfictl adv create`
- `wolfictl adv db`
- `wolfictl adv discover`
- `wolfictl adv list`
- `wolfictl adv migrate` (new)
- `wolfictl adv validate`
- `wolfictl scan`

Planned follow-ups after this PR:

- Ability to add alias information to advisories
- Documentation around how to perform future schema updates

### CODE REVIEW

**Review by luhring — COMMENTED** [review]
(from live review w/ @pdeslaur)

File: pkg/configs/advisory/v2/event.go
Comment (pdeslaur) [inline pkg/configs/advisory/v2/event.go:116]: I generally don't like having `Validate` needing a parameter to support a style of calling the method. The current implementation assumes that `Validate` would be called in a `loop`, using an index. I ended up here after looking at the `Validate` method on `pkg/advisory/request.go` and wondering why we provided `0`!

So - totally not a big deal, but if we could lift `fmt.Sprintf("(index %d)", eventIndex)` out of this method, the code would be cleaner!

File: pkg/configs/rwfs/os/memfs/memfs.go
Comment (pdeslaur) [inline pkg/configs/rwfs/os/memfs/memfs.go:1]: Didn't really review this file - but I'm assuming it works :)

Is that written from scratch or taken from somewhere?

File: pkg/scan/filter.go
Comment (pdeslaur) [inline pkg/scan/filter.go:72]:
```suggestion
			if ok && adv.ResolvedAtVersion(result.TargetAPK.Version) {
				return false
			}
```

File: pkg/scan/filter.go
Comment (pdeslaur) [inline pkg/scan/filter.go:46]:
```suggestion
			// If the advisory contains any events, filter it out!
			if ok && len(adv.Events) >= 1 {
				return false
			}
```

File: pkg/scan/filter.go
Comment (pdeslaur) [inline pkg/scan/filter.go:67]: My 2c: it'd be cool to have a method `GetByAlias` or something like that. It could return a match if the vulnerability ID or an alias matches.

File: pkg/advisory/export_test.go
Comment (pdeslaur) [inline pkg/advisory/export_test.go:65]: Do we still need `TestExportYAML`? Looks like it's folded in a really nice way into the method above!

File: pkg/advisory/request.go
Comment (pdeslaur) [inline pkg/advisory/request.go:43]: I left a comment elsewhere but was confused about the `0` of `Validate(0)`

File: pkg/cli/components/advisory/prompt/model.go
Comment (pdeslaur) [inline pkg/cli/components/advisory/prompt/model.go:50]: Should we use `vuln.ValidateID`, or do we specifically not want to allow GO entries

File: pkg/configs/advisory/v2/advisory.go
Comment (pdeslaur) [inline pkg/configs/advisory/v2/advisory.go:36]: I would specifically document that we're sorting in-place - and calling this method will have side effects!

File: pkg/configs/advisory/v2/error.go
Comment (pdeslaur) [inline pkg/configs/advisory/v2/error.go:10]: Is the comment meaning to say `labeledError implements the error interface`?

File: pkg/configs/rwfs/os/memfs/memfs.go
Comment (luhring) [reply pkg/configs/rwfs/os/memfs/memfs.go:1]: Written from scratch by ChatGPT and then massaged by me :D

File: pkg/configs/advisory/v2/advisory.go
Comment (luhring) [reply pkg/configs/advisory/v2/advisory.go:36]: Unless you object, I'm going to update the method to ensure the slice is copied before sorting happens so there are no side effects!

File: pkg/advisory/advisory.go
Comment (luhring) [reply pkg/advisory/advisory.go:7]: This whole file can be deleted now!

File: pkg/scan/filter.go
Comment (luhring) [reply pkg/scan/filter.go:67]: IIUC, that's the intent of this method (renamed to drop the `ID` suffix):

```go
// GetByVulnerability returns the advisory that references the given
// vulnerability ID as its advisory ID or as one of the advisory's aliases. If
// such an advisory does not exist, the second return value will be false;
// otherwise it will be true.
```

File: pkg/advisory/export_test.go
Comment (luhring) [reply pkg/advisory/export_test.go:65]: Great catch!

File: pkg/cli/components/advisory/prompt/model.go
Comment (luhring) [reply pkg/cli/components/advisory/prompt/model.go:50]: Good question. We don't have good reason to disallow GO- entries, we just haven't used them yet. But we're already using alternative IDs (i.e. GHSA), so we might as well allow GO here, too. I'll update this to use `vuln.ValidateID`.

Comment (luhring) [reply pkg/cli/components/advisory/prompt/model.go:50]: oooo this ends up collapsing this stanza down to:

```go
var validCVEID = field.TextValidationRule(vuln.ValidateID)
```

🤩

File: pkg/configs/advisory/v2/error.go
Comment (luhring) [reply pkg/configs/advisory/v2/error.go:10]: Yeah, basically. It's a doc comment, so the first word matches the method name. Adding this method implements the `error` interface on `labeledError`'s behalf. Let me try an update for clarity...

File: pkg/configs/advisory/v2/advisory.go
Comment (pdeslaur) [reply pkg/configs/advisory/v2/advisory.go:36]: Works for me!

File: pkg/configs/advisory/v2/error.go
Comment (pdeslaur) [reply pkg/configs/advisory/v2/error.go]: Ohhh, right. Yeah, that's a standard golang thing - I forgot.

File: pkg/scan/filter.go
Comment (pdeslaur) [reply pkg/scan/filter.go]: Oh neat - you were one step ahead of me 😎

**Review by pdeslaur — APPROVED** [review]

**Review by pdeslaur — DISMISSED** [review]

**Review by pdeslaur — COMMENTED** [review]
Such green! I'm good to go. I'll wait until Monday to approve to ensure we don't merge by accident over the weekend!

So pumped about this!

**Review by pdeslaur — APPROVED** [review]

**Review by pdeslaur — APPROVED** [review]

**Pull request merged**
